system/lib/wasmfs/file_table.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+// This file defines the open file table of the new file system.
+// Current Status: Work in Progress.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#include "file_table.h"
+
+std::vector<std::shared_ptr<OpenFileDescriptor>> FileTable::entries;
+
+std::shared_ptr<File>& OpenFileDescriptor::getFile() { return file; }
```

Comment:
`File` should be `Lockable<std::unique_ptr<File>>` here. Unfortunately the unique ptr is necessary because `Lockable<File>` cannot store subtypes of `File`. This extra indirection would not be necessary with the original pattern of having Handles as inner classes, so this is a tradeoff we have not considered.